src/Search/Provider/SQLProvider.php

```diff
@@ -1800,8 +1804,10 @@ public static function getWhereCriteria($nott, $itemtype, $ID, $searchtype, $val
             // Add NULL if $val = 0 and not negative search
             // Or negative search on real value
             if (
-                (!$nott && ($val == 0))
-                || ($nott && ($val != 0))
+                ($inittable !== \Entity::getTable())
```

Comment:
Could you explain this change?

Reply:
The root entity ID = 0.
In the KB search, when we searched for the target entity root, the results also included articles that didn't target any entities, so the result was false.

Reply:
It seems correct indeed.